platforms/documentation/docs/src/docs/userguide/authoring-builds/build_environment.adoc

```diff
@@ -131,54 +129,82 @@ spring-boot.version = 2.2.1.RELEASE
+Project properties can be retrieved:
 
-For example, the project property `dryRun`, in the build file below, enables a specific test mode:
+- Lazily: `providers.gradleProperty()`
+- Eagerly: `project.ext()`, `project.findProperty()`, `project.property()`, `project.('<propertyName>')`
 
-.build.gradle
-[source,groovy]
-----
-test {
-  predictiveSelection {
-    enabled = true
-      dryRun = true
-  }
-}
-----
+If a project property is referenced but does not exist, an exception may be thrown and the build will fail.
+You should check for the existence of a project property before directly accessing it with `project.('<propertyName>')` by using the link:{groovyDslPath}/org.gradle.api.Project.html#org.gradle.api.Project:hasProperty(java.lang.String)[`project.hasProperty()`] method.
 
 === Setting a project property
 
-You have three options to add project properties, listed in order of priority:
+You have four options to add project properties, listed in order of priority:
 
 1. *Command Line:* You can add project properties directly to your link:{groovyDslPath}/org.gradle.api.Project.html[Project] object via the `-P` command line option.
 +
 [source,text]
 ----
-$ ./gradlew test -Pgradle.testselection.dryRun
+$ ./gradlew build -PmyProperty='Hi, world'
 ----
-2. *As a System Property:* Gradle creates specially-named system properties for project properties which you can set using the `-D` command line flag or `gradle.properties` file. For the project property `prop`, the system property created is called `org.gradle.project.__prop__`.
+2. *As a System Property:* Gradle creates specially-named system properties for project properties which you can set using the `-D` command line flag or `gradle.properties` file. For the project property `myProperty`, the system property created is called `org.gradle.project.__prop__`.
 +
 [source,text]
 ----
-$ ./gradlew build -Dorg.gradle.project.prop='Hi, world'
+$ ./gradlew build -Dorg.gradle.project.myProperty='Hi, world'
 ----
 +
 .gradle.properties
 [source,properties]
 ----
-org.gradle.project.prop='Hi, world'
+org.gradle.project.myProperty='Hi, world'
 ----
-3. *Environment Variables:* You can set project properties with environment variables. If the environment variable name looks like `ORG_GRADLE_PROJECT___prop__=somevalue`, then Gradle will set a `prop` property on your project object, with the value of `somevalue`.
+3. *Gradle Properties File:* You can also set project properties in `gradle.properties` files.
 +
+.gradle.properties
 [source,text]
 ----
-$ export ORG_GRADLE_PROJECT_myProperty=somevalue
+myProperty='Hi, world'
+----
+4. *Environment Variables:* You can set project properties with environment variables. If the environment variable name looks like `ORG_GRADLE_PROJECT___prop__=somevalue`, then Gradle will set a `prop` property on your project object, with the value of `somevalue`.
```

Comment:
```suggestion
4. *Environment Variables:* You can set project properties with environment variables. If the environment variable name looks like `ORG_GRADLE_PROJECT___myProperty__='Hi, world'`, then Gradle will set a `myProperty` property on your project object, with the value of `Hi, world`.
```

Reply:
I love that you aligned all the examples, by the way! 👏

Reply:
It looks like you missed adding this one. All other examples use `myProperty=Hi, world` except this one.